#VRML_SIM R2021a utf8
WorldInfo {
  info [
    ""
  ]
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -0.39000799213299847 0.6931662976989987 0.6061470529555102 2.2924809387141996
  position 0.4503781724485665 0.8099958219652111 -0.11026090254386281
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  name "maze"
  floorSize 1.28 1.28
  floorTileSize 0.32 0.32
  wallThickness 0.005
  wallHeight 0.05
}
DEF wall Solid {
  translation 0.08 0.0249843 -2.28318e-05
  rotation 3.500134656280128e-11 1 -2.311007011150907e-10 -9.449355117433773e-05
  children [
    DEF wall Shape {
      appearance PBRAppearance {
        baseColor 0.4 0.4 0.4
        metalness 0
      }
      geometry Box {
        size 0.16 0.05 0.005
      }
    }
  ]
  boundingObject USE wall
}
DEF wall Solid {
  translation 0.24 0.0249843 -7.70004e-06
  rotation 3.500134656280128e-11 1 -2.311007011150907e-10 -9.449355117433773e-05
  children [
    DEF wall Shape {
      appearance PBRAppearance {
        baseColor 0.4 0.4 0.4
        metalness 0
      }
      geometry Box {
        size 0.16 0.05 0.005
      }
    }
  ]
  name "solid(1)"
  boundingObject USE wall
}
DEF wall Solid {
  translation 0.240016 0.0249843 -0.160008
  rotation 3.500134656280128e-11 1 -2.311007011150907e-10 -9.449355117433773e-05
  children [
    DEF wall Shape {
      appearance PBRAppearance {
        baseColor 0.4 0.4 0.4
        metalness 0
      }
      geometry Box {
        size 0.16 0.05 0.005
      }
    }
  ]
  name "solid(2)"
  boundingObject USE wall
}
DEF wall Solid {
  translation 0.400016 0.0249843 -0.159992
  rotation 3.500134656280128e-11 1 -2.311007011150907e-10 -9.449355117433773e-05
  children [
    DEF wall Shape {
      appearance PBRAppearance {
        baseColor 0.4 0.4 0.4
        metalness 0
      }
      geometry Box {
        size 0.16 0.05 0.005
      }
    }
  ]
  name "solid(3)"
  boundingObject USE wall
}
DEF wall Solid {
  translation 0.0800153 0.0249843 -0.160022
  rotation 3.500134656280128e-11 1 -2.311007011150907e-10 -9.449355117433773e-05
  children [
    DEF wall Shape {
      appearance PBRAppearance {
        baseColor 0.4 0.4 0.4
        metalness 0
      }
      geometry Box {
        size 0.16 0.05 0.005
      }
    }
  ]
  name "solid(4)"
  boundingObject USE wall
}
DEF wall Solid {
  translation 0.560016 0.0249843 -0.159976
  rotation 3.500134656280128e-11 1 -2.311007011150907e-10 -9.449355117433773e-05
  children [
    DEF wall Shape {
      appearance PBRAppearance {
        baseColor 0.4 0.4 0.4
        metalness 0
      }
      geometry Box {
        size 0.16 0.05 0.005
      }
    }
  ]
  name "solid(5)"
  boundingObject USE wall
}
DEF wall Solid {
  translation 0.56 0.0249843 2.4e-05
  rotation 3.500134656280128e-11 1 -2.311007011150907e-10 -9.449355117433773e-05
  children [
    DEF wall Shape {
      appearance PBRAppearance {
        baseColor 0.4 0.4 0.4
        metalness 0
      }
      geometry Box {
        size 0.16 0.05 0.005
      }
    }
  ]
  name "solid(6)"
  boundingObject USE wall
}
DEF wall Solid {
  translation 0.4 0.0249843 8.00004e-06
  rotation 3.500134656280128e-11 1 -2.311007011150907e-10 -9.449355117433773e-05
  children [
    DEF wall Shape {
      appearance PBRAppearance {
        baseColor 0.4 0.4 0.4
        metalness 0
      }
      geometry Box {
        size 0.16 0.05 0.005
      }
    }
  ]
  name "solid(7)"
  boundingObject USE wall
}
DEF wall Solid {
  translation -0.239986 0.0249843 -0.160053
  rotation 3.500134656280128e-11 1 -2.311007011150907e-10 -9.449355117433773e-05
  children [
    DEF wall Shape {
      appearance PBRAppearance {
        baseColor 0.4 0.4 0.4
        metalness 0
      }
      geometry Box {
        size 0.16 0.05 0.005
      }
    }
  ]
  name "solid(8)"
  boundingObject USE wall
}
DEF wall Solid {
  translation -0.079985 0.0249843 -0.160037
  rotation 3.500134656280128e-11 1 -2.311007011150907e-10 -9.449355117433773e-05
  children [
    DEF wall Shape {
      appearance PBRAppearance {
        baseColor 0.4 0.4 0.4
        metalness 0
      }
      geometry Box {
        size 0.16 0.05 0.005
      }
    }
  ]
  name "solid(9)"
  boundingObject USE wall
}
DEF wall Solid {
  translation -0.0800147 0.0249843 0.159963
  rotation -3.49942e-11 -1 2.31054e-10 9.53072e-05
  children [
    DEF wall Shape {
      appearance PBRAppearance {
        baseColor 0.4 0.4 0.4
        metalness 0
      }
      geometry Box {
        size 0.16 0.05 0.005
      }
    }
  ]
  name "solid(10)"
  boundingObject USE wall
}
DEF wall Solid {
  translation -0.159991 0.0249843 -0.080045
  rotation -9.26462e-15 1 -1.25719e-14 -1.570895307179586
  children [
    DEF wall Shape {
      appearance PBRAppearance {
        baseColor 0.4 0.4 0.4
        metalness 0
      }
      geometry Box {
        size 0.16 0.05 0.005
      }
    }
  ]
  name "solid(11)"
  boundingObject USE wall
}
DEF wall Solid {
  translation 0.0799809 0.0249843 0.159981
  rotation 3.43305e-11 1 -2.26669e-10 -9.530717958661228e-05
  children [
    DEF wall Shape {
      appearance PBRAppearance {
        baseColor 0.4 0.4 0.4
        metalness 0
      }
      geometry Box {
        size 0.16 0.05 0.005
      }
    }
  ]
  name "solid(12)"
  boundingObject USE wall
}
DEF wall Solid {
  translation -0.399986 0.0249843 -0.160069
  rotation 3.500134656280128e-11 1 -2.311007011150907e-10 -9.449355117433773e-05
  children [
    DEF wall Shape {
      appearance PBRAppearance {
        baseColor 0.4 0.4 0.4
        metalness 0
      }
      geometry Box {
        size 0.16 0.05 0.005
      }
    }
  ]
  name "solid(13)"
  boundingObject USE wall
}
DEF wall Solid {
  translation -0.559986 0.0249843 -0.160085
  rotation 3.500134656280128e-11 1 -2.311007011150907e-10 -9.449355117433773e-05
  children [
    DEF wall Shape {
      appearance PBRAppearance {
        baseColor 0.4 0.4 0.4
        metalness 0
      }
      geometry Box {
        size 0.16 0.05 0.005
      }
    }
  ]
  name "solid(14)"
  boundingObject USE wall
}
DEF wall Solid {
  translation -0.160006 0.0249843 0.079955
  rotation -1.25731e-14 1 9.2655e-15 1.5707
  children [
    DEF wall Shape {
      appearance PBRAppearance {
        baseColor 0.4 0.4 0.4
        metalness 0
      }
      geometry Box {
        size 0.16 0.05 0.005
      }
    }
  ]
  name "solid(15)"
  boundingObject USE wall
}
DEF wall Solid {
  translation -0.56 0.0249843 -8.09471e-05
  rotation 3.500134656280128e-11 1 -2.311007011150907e-10 -9.449355117433773e-05
  children [
    DEF wall Shape {
      appearance PBRAppearance {
        baseColor 0.4 0.4 0.4
        metalness 0
      }
      geometry Box {
        size 0.16 0.05 0.005
      }
    }
  ]
  name "solid(16)"
  boundingObject USE wall
}
Robot {
  translation 0.578523 0.0102494 -0.0803093
  rotation 0.018689602688460667 0.9996501437976043 -0.018716002692258253 1.57255
  children [
    LED {
      translation 0 0.001 0.03
      children [
        Shape {
          appearance PBRAppearance {
            metalness 0
          }
          geometry Capsule {
            height 0.01
            radius 0.005
          }
        }
      ]
    }
    Compass {
      children [
        DEF compass Shape {
          appearance PBRAppearance {
            baseColor 0.5 0.6 0.2
            metalness 0
          }
          geometry Box {
            size 0.009 0.009 0.009
          }
        }
      ]
      boundingObject USE compass
      physics Physics {
      }
    }
    DEF body Shape {
      appearance PBRAppearance {
        baseColor 0.1 0.4 0.1
        metalness 0
      }
      geometry Cylinder {
        height 0.001
        radius 0.05
      }
    }
    Solid {
      translation 0 -0.008 -0.04
      children [
        DEF caster_wheel Shape {
          appearance PBRAppearance {
          }
          geometry Capsule {
            height 0.011
            radius 0.005
          }
        }
      ]
      name "solid(2)"
      boundingObject USE caster_wheel
      physics Physics {
      }
    }
    DistanceSensor {
      translation 0 0 -0.05
      rotation 0 1 0 1.5707996938995747
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 0 0.8 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_fwd"
      boundingObject USE sensor
      physics Physics {
      }
    }
    DistanceSensor {
      translation -0.03 0 -0.03
      rotation 0 1 0 3.141592653589793
      children [
        USE sensor
      ]
      name "ds_lft"
      boundingObject USE sensor
      physics Physics {
      }
    }
    DistanceSensor {
      translation 0.03 0 -0.03
      children [
        USE sensor
      ]
      name "ds_rgt"
      boundingObject USE sensor
      physics Physics {
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 817.5335839508841
        axis -1 0 0
        anchor 0.0400001 -1.16495e-05 0.000194469
      }
      device [
        PositionSensor {
          name "ps_rgt"
        }
        RotationalMotor {
          name "rgt_mtr"
        }
      ]
      endPoint Solid {
        translation 0.03999999999994813 -1.173164290095529e-05 0.00019429083845072374
        rotation 0.6404213432761742 0.6404234996387307 0.4239318862586549 3.943521429155956
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 0 0 0
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.02
            }
          }
        ]
        boundingObject USE wheel
        physics DEF wheel_ph Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 658.4260330761992
        axis -1 0 0
        anchor -0.0400001 -1.16987e-05 0.000194411
      }
      device [
        PositionSensor {
          name "ps_lft"
        }
        RotationalMotor {
          name "lft_mtr"
        }
      ]
      endPoint Solid {
        translation -0.04000000000005155 -1.167715432043643e-05 0.00019458048910762988
        rotation -0.6998870701009973 0.6998894241635595 -0.14252327195740078 3.4247314087505294
        children [
          USE wheel
        ]
        name "solid(1)"
        boundingObject USE wheel
        physics USE wheel_ph
      }
    }
  ]
  boundingObject USE body
  physics Physics {
  }
  controller "mazeRunner"
}
